packages/core/src/content/ipfsprovider.ts

```diff
@@ -29,10 +29,10 @@ export class IPFSProvider implements ContentProvider {
     });
   }
 
-  public async put(content: string): Promise<Uri> {
-    return new Promise<Uri>((resolve, reject) => {
+  public async put(content: string): Promise<ContentHeader> {
+    return new Promise<ContentHeader>((resolve, reject) => {
       ipfs.add(Buffer.from(content), (err: any, ipfsHash: any) => {
-        resolve(this.scheme() + "://" + ipfsHash[0].path);
+        resolve({ uri: this.scheme() + "://" + ipfsHash[0].path, hash: ipfsHash });
```

Comment:
I think ipfsHash here is actually poorly named. ipfsHash[0].path gets the actual hash, I believe